modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -539,7 +584,7 @@ namespace cv {
                 net->channels = getParam(net_params, "channels", 3);
                 CV_Assert(net->width > 0 && net->height > 0 && net->channels > 0);
 
-                int current_channels = net->channels;
+                MatShape tensor_shape{ net->channels, net->width, net->height };
```

Comment:
Please replace to `MatShape tensor_shape = shape(net->channels, net->width, net->height);` to keep it compatible with C++98.